src/cxp/controller.ts

```diff
@@ -177,12 +177,17 @@ function environmentFilter(
  */
 export function createController<S extends ConfigurationSubject, C = Settings>(
     context: Context<S, C>,
-    createMessageTransports: (extension: CXPExtensionWithManifest, options: ClientOptions) => Promise<MessageTransports>
+    createMessageTransports: (
+        extension: CXPExtensionWithManifest,
+        options: ClientOptions
+    ) => Promise<MessageTransports>,
+    clientOptions: Partial<ClientOptions>
```

Comment:
These callbacks should probably be handled in the same way that the showMessageRequests and showInputRequests are handled. There should not be 2 different APIs for them (one using observables in Controller instance properties, one using properties in the ClientOptions) because they are both essentially the same thing w.r.t. their data flow.

Also, doing it this way means you don't need this PR, because you don't need to be able to set client options. Not having the ability to set client options is good because it means there is one less place that API consumers might need to configure.

Reply:
Oh I see now that `showMessageRequests` is an `Observable` of callbacks, and since e-c-c's `createController` returns a `cxp.Controller`, this package doesn't need to be changed.